test-basic/concurrent-requests.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var testconfig = require('../etc/test-config.js');
+
+var marklogic = require('../');
+
+var dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+
+describe('concurrent-requests test', function(){
+
+
+    it('should write 1000 documents concurrently', function(done){
+        for(var i=0; i<3; i++) {
+            writeBatch(i,0);
+        }
+
+        done();
```

Comment:
Part of the issue might be on line 31.  Currently done() gets called after the loop finishes starting the 3 writers, but done() should actually be called after after the last batch is written.

Unlike most Java, the call stack in Node.js doesn't reflect the sequence of execution.  That is, the response callbacks execute after the code that passes in the callbacks finishes.

I'd suggest passing the done function reference as another parameter to writeBatch().